pkg/target/namespaces_test.go

```diff
@@ -93,3 +94,244 @@ rules:
   - get
 `)
 }
+
+// This serie of constant is used to prove the need of
+// the namespace field in the objref field of the var declaration.
+// The following tests demonstrates that it creates umbiguous variable
+// declaration if two entities of the kind with the same name
+// but in different namespaces are declared.
+// This is tracking the following issue:
+// https://github.com/kubernetes-sigs/kustomize/issues/1298
+const namespaceNeedInVarMyApp string = `
+resources:
+- elasticsearch-test-service.yaml
```

Comment:
This test case can be changed to a different layout
- base with var
- overlay for test namespace
- overlay for dev namespace
- both
This layout currently fails, but I think we should fix it.

Reply:
@Liujingfang1 Don't completely understand the request. The PR has two tests. The first one is matching your "base with var". The second called workaround is matching "both".

Reply:
This layout is a different use case that came into my mind when reviewing this PR. It is not due to the same root cause, but I think we need to fix it.